src/coreclr/src/jit/lsra.cpp

```diff
@@ -4435,7 +4435,24 @@ void LinearScan::spillGCRefs(RefPosition* killRefPosition)
         {
             continue;
         }
-        unassignPhysReg(regRecord, assignedInterval->recentRefPosition);
+        bool needsKill = varTypeIsGC(assignedInterval->registerType);
+        if (!needsKill)
+        {
+            // The importer will assign a GC type to the rhs of an assignment if the lhs type is a GC type,
+            // even if the rhs is not. See the CEE_STLOC* case in impImportBlockCode(). As a result,
+            // we can have an integer type lclVar that behaves as a GC type. In that case we need to
+            // kill the local.
```

Comment:
>  In that case we need to kill the local

Why? Ideally comments should explain why something is done, not that it's supposed to be done.

Reply:
This is because 1) we can't have any gc refs in registers (even callee-saves) during a pinvoke, 2) the emitter will mark the register as having a GC type if the node has a GC type, and 3) then the code generator asserts that there are no GC types in registers (even callee-save) at the pinvoke prolog.

I'll expand the comment - also it probably should say "spill" the local, even though the code describes it as "killing" gc refs.